### DISCUSSION TITLE
What is the source of the data?

### DISCUSSION THREAD

**Discussion by l3onh4rd** [theOehrly/Fast-F1#634] [General]
Hi, I just dicovered this awesome repository and the library. I am curious about where do you get the data from? What is the source? Is it somehow official from the FIA?

**Comment by theOehrly** [comment]
The data is a combination of multiple sources. Most of the data is taken from the F1 livetiming API (the API that powers their timing and telemetry information in the F1 app). Additional data is taken from ergast.com. Ergast gets most of its data from the official FIA timing documents, with some other sources and manual corrections.
Note that FastF1 partially implements its own data corrections and data augmentation where this is reasonably possible. For example, aligning timestamps of laps between multiple drivers using the gap to leader data, or calculating missing sector times if at least the lap time and two other sector times are known. For some data, there are conditional fallbacks where data is computed from other data if it is not directly available from a supported API.
This makes it somewhat messy to tell the source of the data.

Also, to answer your other question, no it is not really possible to get "the whole dataset". First, because there are multiple input data sets, and second because the data that FastF1 provides to you is a modified and merged form of the input data.

**Reply by l3onh4rd** [reply]
Hi, thanks for your answer. Is the F1 livetiming API somehow accessable? I cannot find any information on such an API.

**Reply by theOehrly** [reply]
Nope, there is no information

**Reply by theOehrly** [reply]
You can try https://openf1.org. That is a completely unrelated project from FastF1 but maybe what you are looking for.

**Reply by l3onh4rd** [reply]
Thanks for the hint. I will have a look into it.

Am I right that you don't know how this library gets the data from f1 livetiming API? Or did I misunderstood you?

I mean do you pay for the data access or anything?

**Reply by theOehrly** [reply]
Obviously I know how FastF1 gets information from that API (and how OpenF1 does it, because it's the same). 
The API itself does not require authentication and the data is not encrypted in any form. Therefore, you can simply request the data from it. I just said that there is no information about this available.

**Reply by l3onh4rd** [reply]
Thanks for your Information!

**Comment by rilhia** [comment]
I have been playing around with this and think it is awesome work guys! However, I have a question about the legality of using the data. I produce tutorial content for data analysis products. My "thing" is developing content that moves away from dry business examples to something that can engage developers personally. The data that is accessible here is perfect for that. I have been hunting around for licensing information, etc, but so far have not found anything which tells me whether I can use this or not. Does anyone have an answer as to whether I could make use of this data for a few tutorial blogs, etc?
